### ISSUE TITLE
Pipenv is unable to build v2.8.0 due to scramp version conflict

### ISSUE THREAD

**Issue by dnltsk** [awslabs/aws-data-wrangler#704]
**Describe the bug**

`awswrangler` v 2.8.0 in Pipfile results in version conflict of transitive dependency `scramp`

**Environment**

My `Pipfile`

```
[[source]]
name = "pypi"
url = "https://pypi.python.org/simple"
verify_ssl = true

[dev-packages]
awswrangler = "==2.8.0"

[packages]
pyspark = "*"

[requires]
python_version = "3.7"

[pipenv]
allow_prereleases = true
```

**To Reproduce**

`pipenv install -d`

Error output is

```
Pipfile.lock not found, creating...
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed!
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/resolver.py", line 741, in _main
[ResolutionFailure]:       resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages, dev)
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/resolver.py", line 709, in resolve_packages
[ResolutionFailure]:       requirements_dir=requirements_dir,
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/resolver.py", line 692, in resolve
[ResolutionFailure]:       req_dir=requirements_dir
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/utils.py", line 1403, in resolve_deps
[ResolutionFailure]:       req_dir=req_dir,
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/utils.py", line 1108, in actually_resolve_deps
[ResolutionFailure]:       resolver.resolve()
[ResolutionFailure]:   File "/Users/daniel.teske/.pyenv/versions/3.8.5/lib/python3.8/site-packages/pipenv/utils.py", line 833, in resolve
[ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: Could not find a version that matches scramp<1.3.0,==1.4.0,>=1.2.0 (from pg8000==1.19.5->awswrangler==2.8.0->-r /var/folders/_k/_q2q_p6d5pl82hh8c3t8c1bd_9_w_x/T/pipenvgjd5neiqrequirements/pipenv-880co3df-constraints.txt (line 2))
Tried: 0.0.0, 0.0.0, 1.0.0, 1.0.0, 1.1.0, 1.1.0, 1.1.1, 1.1.1, 1.2.0, 1.2.0, 1.2.1, 1.2.1, 1.2.2, 1.2.2, 1.3.0, 1.3.0, 1.4.0, 1.4.0
There are incompatible versions in the resolved dependencies:
  scramp<1.3.0,>=1.2.0 (from redshift-connector==2.0.880->awswrangler==2.8.0->-r /var/folders/_k/_q2q_p6d5pl82hh8c3t8c1bd_9_w_x/T/pipenvgjd5neiqrequirements/pipenv-880co3df-constraints.txt (line 2))
  scramp==1.4.0 (from pg8000==1.19.5->awswrangler==2.8.0->-r /var/folders/_k/_q2q_p6d5pl82hh8c3t8c1bd_9_w_x/T/pipenvgjd5neiqrequirements/pipenv-880co3df-constraints.txt (line 2))
```

*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*

**Comment by jaklan** [comment]
Created issues for both dependencies, let's see who fixes that first 😉

**Comment by jaidisido** [comment]
Thank you for raising the respective issues @jaklan, we hope they will be resolved soon. In the meantime, you can circumvent the issue by installing an earlier version of the redshift-connector:
```
pipenv install redshift_connector==2.0.874 awswrangler
```

**Comment by jaklan** [comment]
Should be there soon 🙂 

>Hi @jaklan, thank you for reaching out about this issue. The team is working on a fix for this issue and plans to make a release this coming Monday, May 24th.

_Originally posted by @Brooke-white in https://github.com/aws/amazon-redshift-python-driver/issues/34#issuecomment-844242750_

PS @jaidisido are you sure this workaround works? `redshift_connector==2.0.874` also depends on `scramp<1.3.0`. I'd rather try to pin `pg8000` to `1.18.0` as it depends on `scramp==1.2.2` (if it doesn't break any other dependencies, I can't verify that at the moment).

**Comment by jaidisido** [comment]
I have tested the pipenv installation with the lower version of `redshift_connector` without issues. But you are right that pinning `pg8000` to another version would work just as well. In any case it's just a temporary fix until the dependencies are resolved

**Comment by Brooke-white** [comment]
Hi all,

I've released a new version of redshift_connector, v2.0.881, which addresses the scramp dependency issue. We've enabled dependabot for redshift_connector which should prevent issues like this happening in the future. We appreciate your patience in identifying and working around this issue!

**Comment by jaidisido** [comment]
I can confirm that this new redshift-connector release has solved the issue. Many thanks for the blazing fast response @Brooke-white!
